### ISSUE TITLE
Deprecated bdist_egg uploaded to PyPI

### ISSUE THREAD

**Issue by mr-c** [DataBiosphere/toil#4511]
I received the following email from pypi.org

>### What?
> Since January 1, 2023, we noticed upload(s) of `.egg`/`bdist_egg` distribution file(s) to PyPI for toil.
>
> PyPI is in the process of deprecating `bdist_egg` distribution file uploads, and will not accept them beginning August 1, 2023.
> ### What should I do?
> Update your release tooling to cease upload of `.egg`/`bdist_egg` distribution files. If you are not already, and require binary releases, update your release tooling to upload `.whl`/`bdist_wheel` distribution files instead.
>
> Read more about [Wheels vs Egg](https://packaging.python.org/en/latest/discussions/wheel-vs-egg/) distribution files.

We already publish wheels, and I do see an `.egg` at https://pypi.org/project/toil/#files
![image](https://github.com/DataBiosphere/toil/assets/1330696/a420abd7-3441-4eb1-b9a9-595ae56e7a82)

There is nothing about `.egg`/`bdist_egg` in the source code, but I did see a reference in https://github.com/DataBiosphere/toil/wiki/Toil-Release-Instructions

I recommend that we change those instructions in step 9 to
```
rm -rf dist
pip install pip --upgrade
pip install setuptools wheel build twine --upgrade
python -m build
twine upload dist/*
```
Better yet, this can be a `Makefile` target. 

Thoughts, @DailyDreaming @adamnovak ?

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/TOIL-1357)
┆Issue Number: TOIL-1357

**Comment by adamnovak** [comment]
I got this message too. I followed the release instructions on the wiki for the last release, more or less to the letter. I think previous releases we have done something that doesn't build/upload eggs.

@mr-c If you are confident in your recipe there I think we can go ahead and change the wiki instructions. I have it as a low-priority task to work out what exactly we ought to be doing instead of running setup.py, but unfortunately that means I'll never get to it myself.
